### PR TITLE
Fix power node not connecting to other power nodes

### PR DESCRIPTION
**Pull request by shroobloom**
Fixed a serverside logic bug causing power nodes to not hook up to other power nodes.

### CODE REVIEW

File: lua/entities/ent_jack_powernode/init.lua [hunk 1]
```diff
@@ -68,10 +68,11 @@ function ENT:PhysicsCollide( data, physobj )
         if data.HitEntity:GetClass() == "ent_jack_generator" then
         elseif data.HitEntity:GetClass() == "ent_jack_powernode" then
             -- let the genny do the connecting
-            if self.Generator and IsValid( self.Generator ) and not self.Generator == data.HitEntity.Generator and not table.HasValue( self.Dependents, data.HitEntity ) then
+            if self.Generator and IsValid( self.Generator ) and self.Generator ~= data.HitEntity.Generator and not table.HasValue( self.Dependents, data.HitEntity ) then
                 table.ForceInsert( self.Dependents, data.HitEntity )
                 data.HitEntity.Generator = self.Generator
-
+                print("Connection formed.")
```
Comment (wrefgtzweve) [inline lua/entities/ent_jack_powernode/init.lua]: Might wanna remove this print

Comment (shroobloom) [reply lua/entities/ent_jack_powernode/init.lua]: zamn i knew id miss a print somewhere lol

File: lua/entities/ent_jack_powernode/init.lua [hunk 2]
```diff
@@ -68,10 +68,10 @@ function ENT:PhysicsCollide( data, physobj )
         if data.HitEntity:GetClass() == "ent_jack_generator" then
         elseif data.HitEntity:GetClass() == "ent_jack_powernode" then
             -- let the genny do the connecting
-            if self.Generator and IsValid( self.Generator ) and not self.Generator == data.HitEntity.Generator and not table.HasValue( self.Dependents, data.HitEntity ) then
+            if self.Generator and IsValid( self.Generator ) and self.Generator ~= data.HitEntity.Generator and not table.HasValue( self.Dependents, data.HitEntity ) then
                 table.ForceInsert( self.Dependents, data.HitEntity )
                 data.HitEntity.Generator = self.Generator
-
+
```
Comment (wrefgtzweve) [inline lua/entities/ent_jack_powernode/init.lua]: Left over whitespace

**Comment by wrefgtzweve** [comment]
I don't see how changing `not ==` to `~=` is going to change anything functional about the code

**Comment by periapsises** [comment]
image

**Comment by wrefgtzweve** [comment]
Yeah seems like the statement is interpreted as `( not 1 ) == 0 `, if you fix the whitespace I'll merge it

**Review by wrefgtzweve — APPROVED** [review]

**Pull request merged**